flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/RankITCase.scala

```diff
@@ -1311,4 +1311,49 @@ class RankITCase(mode: StateBackendMode) extends StreamingWithStateTestBase(mode
     assertEquals(expected2.sorted, sink2.getRetractResults.sorted)
   }
 
+  @Test
+  def testRetractRank(): Unit = {
```

Comment:
Could you use `testTopNWithGroupByAvgWithoutRowNumber` to verify this which already have a meaningful testing data?

Reply:
done